src/chat_module.ts

```diff
@@ -184,6 +196,137 @@ export class ChatModule implements ChatInterface {
     return this.getMessage();
   }
 
+  /**
+   * Similar to `generate()`; but instead of using callback, we use an async iterable.
+   * @param request Request for chat completion.
+   * @param genConfig Generation config extraced from `request`.
+   */
+  async* chatCompletionAsyncChunkGenerator(
+    request: ChatCompletionRequestStreaming,
+    genConfig: GenerationConfig
+  ): AsyncGenerator<ChatCompletionChunk, void, void> {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const model = this.currentLocaId!;
+    const created = Date.now();
+    const id = crypto.randomUUID();
+
+    this.interruptSignal = false;
+    await this.prefill(request.messages, genConfig);
+    let prevMessage = "";  // to know where to start slicing the delta
```

Comment:
It might be more memory efficient to just keep a counter for prev length

Reply:
Fixed!